### ISSUE TITLE
PHP unable to call C# method, e.g. set_prefix

### ISSUE THREAD

**Issue by jdluzen** [peachpiecompiler/peachpie#584]
Possibly related to #583, if I create a new class which WordPress should use for the $wpdb, without inheriting from the wpdb class, it cannot find certain methods.

For example, trying to implement `set_prefix` via `public PhpValue set_prefix(PhpValue prefix, PhpValue set_table_names = default/*true*/)`, it throws a `System.NotImplementedException` with
```
   at Pchp.Core.Dynamic.SetFieldBinder.Bind(DynamicMetaObject target, DynamicMetaObject[] args)
   at System.Dynamic.DynamicMetaObjectBinder.Bind(Object[] args, ReadOnlyCollection`1 parameters, LabelTarget returnLabel)
   at System.Runtime.CompilerServices.CallSiteBinder.BindCore[T](CallSite`1 site, Object[] args)
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid3[T0,T1,T2](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at <Root>wp-includes.load_php.wp_set_wpdb_vars(Context <ctx>) in A:\src\wpdotnet-sdk\wordpress\wp-includes\load.php:line 484
   at <Root>.wp-settings_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in A:\src\wpdotnet-sdk\wordpress\wp-settings.php:line 123
   at Pchp.Core.Context.ScriptInfo.Evaluate(Context ctx, PhpArray locals, Object this, RuntimeTypeHandle self)
   at Pchp.Core.Context.Include(String cd, String path, PhpArray locals, Object this, RuntimeTypeHandle self, Boolean once, Boolean throwOnError)
   at <Root>.wp-config_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in A:\src\wpdotnet-sdk\wordpress\wp-config.php:line 89
   at <Root>.wp-config_php.<Main>`0(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>)
   at Pchp.Core.Context.ScriptInfo.Evaluate(Context ctx, PhpArray locals, Object this, RuntimeTypeHandle self)
   at Pchp.Core.Context.Include(String cd, String path, PhpArray locals, Object this, RuntimeTypeHandle self, Boolean once, Boolean throwOnError)
   at <Root>.wp-load_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in A:\src\wpdotnet-sdk\wordpress\wp-load.php:line 37
   at <Root>.wp-load_php.<Main>`0(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>)
   at Pchp.Core.Context.ScriptInfo.Evaluate(Context ctx, PhpArray locals, Object this, RuntimeTypeHandle self)
   at Pchp.Core.Context.Include(String cd, String path, PhpArray locals, Object this, RuntimeTypeHandle self, Boolean once, Boolean throwOnError)
   at <Root>wp-admin.install_php.<Main>(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>) in A:\src\wpdotnet-sdk\wordpress\wp-admin\install.php:line 36
   at <Root>wp-admin.install_php.<Main>`0(Context <ctx>, PhpArray <locals>, Object this, RuntimeTypeHandle <self>)
   at Pchp.Core.Context.ScriptInfo.Evaluate(Context ctx, PhpArray locals, Object this, RuntimeTypeHandle self)
   at Peachpie.AspNetCore.Web.RequestContextCore.ProcessScript(ScriptInfo script)
```

I have attempted to debug it, and try a variety of different method signatures, but nothing has worked.

**Comment by jakubmisek** [comment]
This exception is caused by setting a property that does not exist. It is not related to a method call.

Maybe that helps? We'll improve the error message.

**Comment by jakubmisek** [comment]
@jdluzen please try with latest fixes (you can use builds from appveyor >= `1.0.0-appv3046`)

It should provide more insights on the inaccessible property.

**Comment by jdluzen** [comment]
That was super fast, thanks so much!
It was indeed missing `field_types`. Adding that allowed it to call into `set_prefix` successfully.
